spec/controllers/vm_common_spec.rb

```diff
@@ -143,4 +143,71 @@
+      template_orph = FactoryGirl.create(:template_orphaned)
+      expect(controller.parent_folder_id(vm_orph)).to eq('xx-orph')
+      expect(controller.parent_folder_id(template_orph)).to eq('xx-orph')
+    end
+    it 'returns id of archived folder for archived VM/Template' do
+      vm_arch = FactoryGirl.create(:vm)
+      template_arch = FactoryGirl.create(:template_infra)
+      expect(controller.parent_folder_id(vm_arch)).to eq('xx-arch')
+      expect(controller.parent_folder_id(template_arch)).to eq('xx-arch')
+    end
+    it 'returns id of Availability Zone folder for Cloud VM that has one' do
+      vm_cloud_with_az = FactoryGirl.create(:vm_cloud_with_az)
+      expect(controller.parent_folder_id(vm_cloud_with_az)).to eq(TreeBuilder.build_node_cid(vm_cloud_with_az.availability_zone))
+    end
+    it 'returns id of Provider folder for Cloud VM without Availability Zone' do
+      vm_cloud_without_az = FactoryGirl.create(:vm_cloud_without_az)
+      expect(controller.parent_folder_id(vm_cloud_without_az)).to eq(TreeBuilder.build_node_cid(vm_cloud_without_az.ext_management_system))
+    end
+    it 'returns id of Provider folder for Cloud Template' do
+      template_cloud = FactoryGirl.create(:template_cloud_with_ems)
+      expect(controller.parent_folder_id(template_cloud)).to eq(TreeBuilder.build_node_cid(template_cloud.ext_management_system))
+    end
+    it 'returns id of Provider folder for infra VM/Template without blue folder' do
+      vm_infra = FactoryGirl.create(:vm_infra_no_folder)
+      template_infra = FactoryGirl.create(:template_infra_no_folder)
+      expect(controller.parent_folder_id(vm_infra)).to eq(TreeBuilder.build_node_cid(vm_infra.ext_management_system))
+      expect(controller.parent_folder_id(template_infra)).to eq(TreeBuilder.build_node_cid(template_infra.ext_management_system))
+    end
+    it 'returns id of blue folder for VM/Template with one' do
+      folder = FactoryGirl.create(:ems_folder)
+      vm_infra_folder = FactoryGirl.create(:vm_infra_no_folder)
+      vm_infra_folder.with_relationship_type("ems_metadata") { vm_infra_folder.parent = folder } # add folder
+      template_infra_folder = FactoryGirl.create(:template_infra_no_folder)
+      template_infra_folder.with_relationship_type("ems_metadata") { template_infra_folder.parent = folder } # add folder
+      expect(controller.parent_folder_id(vm_infra_folder)).to eq(TreeBuilder.build_node_cid(folder))
+      expect(controller.parent_folder_id(template_infra_folder)).to eq(TreeBuilder.build_node_cid(folder))
+    end
+  end
+
+  context "#resolve_node_info" do
+    before do
+      role = MiqUserRole.find_by_name("EvmRole-operator")
+      @group = FactoryGirl.create(:miq_group, :miq_user_role => role, :description => "Resolve node info Group")
+      login_as FactoryGirl.create(:user, :userid => 'resolve_node_info_wilma', :miq_groups => [@group], :settings => {})
+      @vm_arch = FactoryGirl.create(:vm)
+      extend VmCommon
```

Comment:
This concerns me at the spec level...could this be a problem?  cc @kbrock @jrafanie ?